### DISCUSSION TITLE
Is there a tutorial for sdxl with only 8GB vram ?

### DISCUSSION THREAD

**Discussion by cgidesign-de** [vladmandic/sdnext#2342] [Q&A] [answered]
My main GPU is in repair, so I need to try sdnext on a notebook.

I read that sdxl should be possible with 8 GB vram but struggle to find a settings guide for this.

I am running latest sdnext from yesterday on Windows 11 with 16 GB ram and an rtx 2070 with 8 GB vram.

Thanks for help.

**Comment by MaxJa4** [comment, marked as answer]
Look here: https://github.com/vladmandic/automatic/wiki/SD-XL#vram-optimization

**Comment by cgidesign-de** [comment]
Thanks, it works.

**Comment by cgidesign-de** [comment]
Nope, I spoke to early.
It only works for the first generation. Trying to get a second image in text2image generates OOM error.
I tried the shuffle method so far.
Only solution is to restart sdnext after each generation.

Error:
CUDA out of memory. Tried to allocate 50.00 MiB. GPU 0 has a total capacty of 8.00 GiB of which 0 bytes is free. Of the allocated memory 7.01 GiB is allocated by PyTorch, and 273.62 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

**Comment by cgidesign-de** [comment]
Another update:
medvram seems to work well.
:-)

**Reply by Tillerz** [reply]
Just a question, which model are you using, is it also around 6-7 GB large? Cause I have a friend who doesn't get SDXL running with 8 GB, but if you did, they she should be able too. Wonder what she's doing wrong.

**Reply by cgidesign-de** [reply]
Models are in the 6 to 7 GB range.
I am not using second pass or highres fix; maybe that is the difference?
Refiner Model is set to NONE in sdnext settings -> Excecution & Models
Cross-attention is set to Scaled-Dot-Product in Compute Settings
I needed to activate medvram in settings: Diffuser settings -> Enable model CPU offload (--medvram)
Size I can generate is 1024x1280
And I am not using add ons like adetailer
Is she really using the Diffuser backend? In Execution & Models at the top "diffusers" must be selected not "original"
There is a vram handling issue with newer Nvidia drivers. According to the posts I read the last "good" driver version is 531.79 - check the top of the page here: https://github.com/vladmandic/automatic/wiki/Installation

Generation with 30 steps and Euler A takes below one minute.

I made a fresh install of sdnext. In the past I had issues when just updating an older installation.

Hope this helps.

**Reply by usigg3** [reply]
Are you generating 1024x1280, 30 steps in about a minute? I haven't tried in a while but last time I tried it took me about 7 minutes, but I only have 6 GB VRAM.

**Reply by cgidesign-de** [reply]
Yes, about a minute:
Notebook with:
CPU: Intel(R) Core(TM) i7-10750H
RAM: 16 GB
GPU: RTX 2070 MaxQ (8 GB vram)
Storage: 512 GB NVMe-SSD
Windows 11 with browser Edge
sdnext: from 2023-10-18 (fresh install)
Sampler: Euler A (others might take longer - e.g. DPM SDE)
